sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -893,8 +893,10 @@ public String getId() {
     public <T> CosmosPagedFlux<T> readAllItems(
         PartitionKey partitionKey,
         Class<T> classType) {
+        CosmosQueryRequestOptions queryRequestOptions = new CosmosQueryRequestOptions();
+        queryRequestOptions.setPartitionKey(partitionKey);
 
-        return this.readAllItems(partitionKey, new CosmosQueryRequestOptions(), classType);
+        return this.readAllItems(partitionKey, queryRequestOptions, classType);
```

Comment:
Not sure I follow... why do we set the partition key twice? looks to me that the called readAllItems() should  either ensure partitionKey == queryRequestOptions.partitionKey or set it if null or empty.

Reply:
Currently we have two public readAllItems API:
 public <T> CosmosPagedFlux<T> readAllItems(PartitionKey partitionKey,Class<T> classType) 

public <T> CosmosPagedFlux<T> readAllItems(
PartitionKey partitionKey, CosmosQueryRequestOptions options,Class<T> classType) 

This change is for the first API, just to make sure we also set the partitionKey in the options. Since the option will be later passed on query pipeline, and used to set the partitionKey header.